src/main/java/uk/gov/hmcts/reform/preapi/repositories/UserRepository.java

```diff
@@ -12,4 +16,30 @@ public interface UserRepository extends SoftDeleteRepository<User, UUID> {
     boolean existsByIdAndDeletedAtIsNull(UUID id);
 
     Optional<User> findByIdAndDeletedAtIsNull(UUID id);
+
+    @Query(
+        """
+        SELECT u FROM User u
+        WHERE u.deletedAt IS NULL
+        AND (:firstName IS NULL OR u.firstName ILIKE %:firstName%)
+        AND (:lastName IS NULL OR u.lastName ILIKE %:lastName%)
+        AND (:email IS NULL OR u.email ILIKE %:email%)
+        AND (:organisation IS NULL OR u.organisation ILIKE %:organisation%)
+        AND (CAST(:courtId as uuid) IS NULL OR EXISTS (SELECT 1 FROM u.appAccess aa WHERE aa.court.id = :courtId))
+        AND (CAST(:roleId as uuid) IS NULL OR EXISTS (SELECT 1 FROM u.appAccess aa WHERE aa.role.id = :roleId))
+        AND (:isPortalUser = false OR EXISTS (SELECT 1 FROM u.portalAccess pa WHERE pa.user = u))
+        AND (:isAppUser = false OR EXISTS (SELECT 1 FROM u.appAccess aa WHERE aa.user = u))
```

Comment:
Might want a deleted check here as well
```suggestion
        AND (:isPortalUser = false OR EXISTS (SELECT 1 FROM u.portalAccess pa WHERE pa.user = u AND pa.deletedAt IS NULL))
        AND (:isAppUser = false OR EXISTS (SELECT 1 FROM u.appAccess aa WHERE aa.user = u and aa.deletedAt IS NULL))
```